src/classes/CON_DeleteContactOverride_CTRL.cls

```diff
@@ -32,60 +32,143 @@
 
-    /** @description The contact id of the record to delete. */
-    private final string conId;
-    private string retURL;
+    /** @description The contact record the user is attempting to delete. */
+    private Contact queryContact;
+    /** @description The id of the contact's account. */
+    private String accountId;
+    /** @description The return URL. */
+    private String retURL;
+
+    /** @description Formatted string prompting the user to delete the account instead. */
+    public String helpDeleteAccountInstead {get;set;}
+    /** @description Flag to tell the page which operation we're doing. */
+    public Boolean shouldDeleteContactAlone {get;set;}
+    /** @description Boolean for disabling the account deletion button if there are errors preventing deletion. */
+    public Boolean hasPageMessages {
+        get {
+            return ApexPages.hasMessages();
+        }
+        set;
+    }
 
     /*******************************************************************************************************
-    * @description Standard controller constructor.
+    * @description Standard controller constructor. Queries for contact, verifies whether the contact or its
+    * account should be deleted.
     */ 
     public CON_DeleteContactOverride_CTRL(ApexPages.StandardController stdController) {
-        conId = stdController.getId();
         retURL = ApexPages.currentPage().getParameters().get('returl');
         
         //if we don't have a return URL, go back to Contacts home.
-        if (string.isBlank(retURL)) {
+        if (String.isBlank(retURL)) {
             retURL = '/003/o';
         }
-    }
 
-    /*******************************************************************************************************
-    * @description Action method in Contact Delete Button override, handles deleting a contact and
-    * potentially its account if it is a system account (1:1, individual, or household) and is the last 
-    * contact in the account.
-    * @return pageReference Redirects to the Contacts tab.
-    */ 
-    public pageReference processDelete() {
-        Contact queryContact = [SELECT Id, AccountId, Account.npe01__SYSTEMIsIndividual__c FROM Contact WHERE Id = :conId];
-        string accId = queryContact.AccountId;
-        boolean shouldDeleteContactAlone = true;
+        queryContact = [SELECT Id, Name, AccountId, Account.Name, Account.npe01__SYSTEMIsIndividual__c FROM Contact WHERE Id = :stdController.getId()];
+
+        helpDeleteAccountInstead = String.format(Label.ConfirmDeleteAccount, new List<String>{queryContact.Name, queryContact.Account.Name});
+
+        shouldDeleteContactAlone = true;
 
         if (queryContact.AccountId != null && queryContact.Account.npe01__SYSTEMIsIndividual__c) {
-            list<AggregateResult> contactsInHousehold = [SELECT COUNT(id) ct FROM Contact WHERE AccountId = :accId GROUP BY AccountId HAVING COUNT(id) = 1];
+            accountId = queryContact.AccountId;
```

Comment:
same issue previously mentioned.  with private contacts, I don't think we can safely dereference  contact.account.foo